### PR TITLE
[Fix #1760] Fix wifi_networks for OS X 10.9

### LINKED ISSUE #1760: wifi_networks table on OS X 10.9.5 (13F1507)
I've just cloned and installed osquery and then tried to do this:

lsanmartin ~/src/c++/osquery >  ./build/darwin10.9/osquery/osqueryi
osquery - being built, with love, at Facebook
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Using a virtual database. Need help, type '.help'
osquery> select * from wifi_networks;
2016-01-11 14:59:26.090 osqueryi[29498:507] -[__NSArrayM getObjects:andKeys:]: unrecognized selector sent to instance 0x7f9153c58740
E0111 14:59:26.091904 1905287952 registry.cpp:336] table registry wifi_networks plugin caused unknown exception

Am I missing something? couldnt find any open issue regarding this.

### PR DESCRIPTION
**Pull request by sharvilshah**
* Take into account slight difference in wifi_networks plist in OS X 10.9 and later.
* Update tests.

### CODE REVIEW

**Comment by osqueryer** [comment]
Can one of the osquery admins reply with "ok to test" to kick off a build...

File: osquery/tables/networking/darwin/tests/wifi_tests.mm [hunk 1]
```diff
@@ -22,27 +23,48 @@
...
+  }
+
+  std::string key = nullptr;
+  std::string path = nullptr;
+  if (qd.front().at("major") == "10" && qd.front().at("minor") == "9") {
+    key = "RememberedNetworks";
+    path = kTestDataPath + "test_airport_darwin_10_9.plist";
+  } else {
+    key = "RememberedNetworks";
+    path = kTestDataPath + "test_airport_darwin_10_9.plist";
```
Comment (theopolis) [inline osquery/tables/networking/darwin/tests/wifi_tests.mm]: Should this be a `test_airport_darwin_10_10`?

File: tools/tests/test_airport_darwin_10_9.plist [hunk 2]
```diff
@@ -0,0 +1,100 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (theopolis) [inline tools/tests/test_airport_darwin_10_9.plist]: Mind shortening the file name?

File: osquery/tables/networking/darwin/tests/wifi_tests.mm (hunk 1 above)
Comment (sharvilshah) [inline osquery/tables/networking/darwin/tests/wifi_tests.mm]: d'oh! copy pasta error, it should be `test_airport.plist` which already exists for 10.10 and later. Do you prefer the longer name?

File: tools/tests/test_airport_darwin_10_9.plist (hunk 2 above)
Comment (sharvilshah) [inline tools/tests/test_airport_darwin_10_9.plist]: Got a preference? There is already a `test_airport.plist` in the tests directory for OS X 10.10 and later.

Comment (theopolis) [inline tools/tests/test_airport_darwin_10_9.plist]: I would keep `test_airport` and express your implicit shame for 10.9 users by calling this `test_airport_legacy.plist` ;)

File: osquery/tables/networking/darwin/wifi.mm [hunk 3]
```diff
@@ -147,15 +146,23 @@ QueryData genKnownWifiNetworks(QueryContext& context) {
...
+          results);
+    }
+  } else if (CFGetTypeID(networks) == CFDictionaryGetTypeID()) {
+    auto count = CFDictionaryGetCount((CFDictionaryRef)networks);
+    const void* keys[count];
+    const void* values[count];
+    CFDictionaryGetKeysAndValues((CFDictionaryRef)networks, keys, values);
+
+    for (CFIndex i = 0; i < count; i++) {
+      parseNetworks((CFDictionaryRef)values[i], results);
```
Comment (theopolis) [inline osquery/tables/networking/darwin/wifi.mm]: I'm wondering-- if this `values[1]`, or the above `networks`, contained something other than a dict-- if the `parseNetworks` code would barf? Might need to make use of some more Apple RTTI and do a paranoid check that these are `CFDictionaryGetTypeID` types. I think you have a guarantee that it's either `nullptr` or some derived `CFTypeRef`.

File: tools/tests/test_airport_darwin_10_9.plist (hunk 2 above)
Comment (sharvilshah) [inline tools/tests/test_airport_darwin_10_9.plist]: >Two hardest things in computer science: cache invalidation, naming things and off by one errors" 

Updating :smile:

File: osquery/tables/networking/darwin/wifi.mm (hunk 3 above)
Comment (sharvilshah) [inline osquery/tables/networking/darwin/wifi.mm]: `parseNetworks` is already doing a `nullpr` check, I could add a check for `CFGetTypeID(network) != CFDictionaryGetTypeID()`

**Comment by sharvilshah** [comment]
Hrm, RocksDB can't get a handle when running the tests locally:

```
3/9 Test #3: osquery_tables_tests .............***Exception: SegFault  0.18 sec
WARNING: Logging before InitGoogleLogging() is written to STDERR
W0113 20:35:27.484890 1913696256 db_handle.cpp:113] [Ref #1629] Not allowed to create DBHandle instance
I0113 20:35:27.486183 1913696256 db_handle.cpp:106] RocksDB: [ERROR] Error when reading /private/tmp/osquery-tests0/unittests.db dir
```

Wondering if it's just my system.

**Comment by theopolis** [comment]
ok to test

**Comment by sharvilshah** [comment]
Same error on the Jenkins box. And just tested master on my system with no problems. Looking...

**Comment by osqueryer** [comment]
Build (ffed82d1e57740ff6ea72a1816d421b03d8a162a) failed :-1:

**Comment by osqueryer** [comment]
Build (ffed82d1e57740ff6ea72a1816d421b03d8a162a) is good! :+1:

**Pull request merged**